app/code/Oleksandrr/Chat/view/frontend/web/js/sidebar.js

```diff
@@ -128,19 +145,35 @@ define([
             $('#oleksandrr-chat-message-field').append(field);
         },
 
+        /**
+         * @param {jsDoc} list
+         */
         showPreviousMessages: function (list) {
-            var context = this;
+            /**
+             * @var {Object} item
+             * @var {String} item.created_at
+             * @var {String} item.author_name
+             * @var {String} item.message
+             */
+
             $.each(list, function (index, item) {
                 var date = new Date(item.created_at);
 
                 $('#oleksandrr-chat-message-field').append(
-                    context.createMessage(item.author_name, date.toLocaleTimeString(), item.message)
+                    this.createMessage(item.author_name, date.toLocaleTimeString(), item.message)
```

Comment:
@pandaUbuntu DOM operations are the most expensive ones. Generate content in the loop, append once. Never append in the loop unless this is really needed (most likely it is not).